beamer/tests/contracts/test_request_manager.py

```diff
@@ -1037,6 +1037,50 @@ def test_challenge_after_l1_resolution(request_manager, token, claim_stake, cont
         request_manager.challengeClaim(claim_id)
 
 
+def test_withdraw_on_behalf(
+    request_manager, token, claim_stake, finality_period, challenge_period_extension
+):
+    first_challenger, second_challenger, requester = alloc_accounts(3)
+    (claimer,) = alloc_whitelisted_accounts(1, {request_manager})
+
+    request_id = make_request(request_manager, token, requester, requester, 1)
+    claim = request_manager.claimRequest(
+        request_id, FILL_ID, {"from": claimer, "value": claim_stake}
+    )
+    claim_id = claim.return_value
+    first_challenger_eth_balance = web3.eth.get_balance(first_challenger.address)
+    second_challenger_eth_balance = web3.eth.get_balance(second_challenger.address)
+
+    # First challenger challenges
+    request_manager.challengeClaim(claim_id, {"from": first_challenger, "value": claim_stake + 1})
+    # Claimer outbids again
+    request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake + 1})
+    # Second challenger challenges
+    request_manager.challengeClaim(claim_id, {"from": second_challenger, "value": claim_stake + 1})
+
+    # Timetravel after claim period
+    chain.mine(timedelta=finality_period + challenge_period_extension)
+
+    request_manager.withdraw(second_challenger, claim_id, {"from": first_challenger})
+
+    # second challenger should have won claim stake which is the excess amount the
+    # claimer put in
+    assert (
+        web3.eth.get_balance(second_challenger.address)
+        == second_challenger_eth_balance + claim_stake
+    )
+
+    with brownie.reverts("Challenger has nothing to withdraw"):
+        request_manager.withdraw(claim_id, {"from": second_challenger})
+
+    request_manager.withdraw(first_challenger, claim_id, {"from": first_challenger})
+
+    assert (
```

Comment:
If second challenger won the game, why first challenger has more assets in the end? This is now kind of saying first challenger won it as he has more assets.

Reply:
Actually both challengers won, because the challengers as a team won against the claimer. It is possible that multiple challengers join a challenge against a claimer. The distribution of the stakes will happen accordingly to the amount they put in. 
The claimer, in this case, won nothing.

[here](https://docs.beamerbridge.com/whitepaper.html#rightful-claims-resolutions) is a detailed description of how multiple  challengers join a challenge and how stakes are distributed. Please ask if oyu have any further questions :)